### PR TITLE
Update CMake Windows support

### PR DESCRIPTION
**Pull request by dnakamura**
- Enable using clang to compile the interpreter
- Update tool wrappers to work with new fixpath

Signed-off-by: Devin Nakamura <devinn@ca.ibm.com>

### CODE REVIEW

**Comment by dnakamura** [comment]
See also eclipse/openj9#11892

**Comment by pshipton** [comment]
Are any of these changes needed for jdk16?

**Comment by pshipton** [comment]
jenkins compile win jdknext

**Comment by pshipton** [comment]
> Are any of these changes needed for jdk16?

Since the build with https://github.com/eclipse/openj9/pull/11892 failed on jdk16, the answer must be yes.
https://ci.eclipse.org/openj9/job/Build_JDK16_x86-64_windows_cm_Personal/2/

File: closed/autoconf/custom-hook.m4 [hunk 1]
```diff
@@ -164,6 +164,9 @@ AC_DEFUN([OPENJ9_CONFIGURE_COMPILERS],
   AC_ARG_WITH(openj9-developer-dir, [AS_HELP_STRING([--with-openj9-developer-dir], [build OpenJ9 with a specific Xcode version])],
     [OPENJ9_DEVELOPER_DIR=$with_openj9_developer_dir],
     [OPENJ9_DEVELOPER_DIR=])
+  if test "x$OPENJDK_BUILD_OS" = xwindows ; then
+     UTIL_REQUIRE_PROGS([OPENJ9_CLANG], [clang])
```
Comment (keithc-ca) [inline closed/autoconf/custom-hook.m4:168]: nit: please fix indentation (2 spaces per level) to be consistent

File: closed/autoconf/custom-hook.m4 [hunk 2]
```diff
@@ -695,26 +698,29 @@ AC_DEFUN([OPENJ9_GENERATE_TOOL_WRAPPER],
...
+  OPENJ9_GENERATE_TOOL_WRAPPER([ml], [$AS])
+  OPENJ9_GENERATE_TOOL_WRAPPER([ml64], [$ML64])
   OPENJ9_GENERATE_TOOL_WRAPPER([rc], [$RC])
-  OPENJ9_GENERATE_TOOL_WRAPPER([mc], [$SDK_BIN_DIR/mc])
+  OPENJ9_GENERATE_TOOL_WRAPPER([mc], [$MC])
   OPENJ9_GENERATE_TOOL_WRAPPER([nasm], [$NASM])
   OPENJ9_GENERATE_TOOL_WRAPPER([java], [$JAVA])
   OPENJ9_GENERATE_TOOL_WRAPPER([jar], [$JAR])
   OPENJ9_GENERATE_TOOL_WRAPPER([javac], [$JAVAC])
+  OPENJ9_GENERATE_TOOL_WRAPPER([clang], [$OPENJ9_CLANG])
```
Comment (keithc-ca) [inline closed/autoconf/custom-hook.m4:725]: Please put these calls in sorted order (like jdk11).

**Review by keithc-ca — CHANGES_REQUESTED** [review]

**Review by pshipton — APPROVED** [review]

File: closed/autoconf/custom-hook.m4 [hunk 3]
```diff
@@ -695,26 +698,29 @@ AC_DEFUN([OPENJ9_GENERATE_TOOL_WRAPPER],
...
-  OPENJ9_GENERATE_TOOL_WRAPPER([lib], [$AR])
-  OPENJ9_GENERATE_TOOL_WRAPPER([link], [$LD])
-  OPENJ9_GENERATE_TOOL_WRAPPER([ml], [$MSVC_BIN_DIR/ml])
-  OPENJ9_GENERATE_TOOL_WRAPPER([ml64], [$MSVC_BIN_DIR/ml64])
-  OPENJ9_GENERATE_TOOL_WRAPPER([rc], [$RC])
-  OPENJ9_GENERATE_TOOL_WRAPPER([mc], [$SDK_BIN_DIR/mc])
-  OPENJ9_GENERATE_TOOL_WRAPPER([nasm], [$NASM])
+  OPENJ9_GENERATE_TOOL_WRAPPER([clang], [$OPENJ9_CLANG])
   OPENJ9_GENERATE_TOOL_WRAPPER([java], [$JAVA])
   OPENJ9_GENERATE_TOOL_WRAPPER([jar], [$JAR])
```
Comment (keithc-ca) [inline closed/autoconf/custom-hook.m4:717]: 'jar' comes before 'java'.

**Review by keithc-ca — APPROVED** [review]

**Comment by keithc-ca** [comment]
> > Are any of these changes needed for jdk16?
> Since the build with [eclipse/openj9#11892](https://github.com/eclipse/openj9/pull/11892) failed on jdk16, the answer must be yes.

@dnakamura Please put together an equivalent change for jdk16.

**Comment by keithc-ca** [comment]
Jenkins test sanity win,wincm jdknext

**Comment by keithc-ca** [comment]
Jenkins test sanity wincm jdknext depends eclipse/openj9#11892

**Review by keithc-ca — CHANGES_REQUESTED** [review]
Apparently changes are required here or in openj9:
```
15:37:50  com\ibm\oti\VMCPTool\Main.java:451: error: cannot find symbol
15:37:50  			return new VirtualMethodRef(e, classes);
15:37:50  			           ^
15:37:50    symbol:   class VirtualMethodRef
15:37:50    location: class Main
```

**Comment by dnakamura** [comment]
Appears to just be a typo introduced when converting the if statements.

**Comment by keithc-ca** [comment]
Jenkins test sanity wincm jdknext depends eclipse/openj9#11892

**Review by keithc-ca — APPROVED** [review]

**Pull request merged**